core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt

```diff
@@ -0,0 +1,250 @@
+import kotlin.system.exitProcess
+
+private const val STATUS = 10
+private const val ZERO = 0
+
+open class ErrorActivity : BaseActivity() {
+  @Inject
+  lateinit var bookDao: NewBookDao
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  @Inject
+  lateinit var mountPointProducer: MountPointProducer
+
+  @Inject
+  lateinit var fileLogger: FileLogger
+
+  private var exception: Throwable? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_kiwix_error)
+    val extras = intent.extras
+    exception = if (extras != null && safeContains(extras)) {
+      extras.getSerializable(EXCEPTION_KEY) as Throwable
+    } else {
+      null
+    }
+    setupReportButton()
+    restartButton.setOnClickListener { restartApp() }
+  }
+
+  private fun setupReportButton() {
+    reportButton.setOnClickListener {
+      startActivityForResult(
+        Intent.createChooser(emailIntent(), "Send email..."), 1
+      )
+    }
+  }
+
+  private fun emailIntent(): Intent {
+    return Intent(Intent.ACTION_SEND).apply {
+      type = "vnd.android.cursor.dir/email"
+      putExtra(
+        Intent.EXTRA_EMAIL,
+        arrayOf("android-crash-feedback@kiwix.org")
+      )
+      putExtra(Intent.EXTRA_SUBJECT, subject)
+      putExtra(Intent.EXTRA_TEXT, buildBody())
+      if (allowLogs.isChecked) {
+        val file = fileLogger.writeLogFile(this@ErrorActivity)
+        val path =
+          FileProvider.getUriForFile(
+            this@ErrorActivity,
+            applicationContext.packageName + ".fileprovider",
+            file
+          )
+        addFlags(android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION)
+        this.putExtra(android.content.Intent.EXTRA_STREAM, path)
+      }
+    }
+  }
+
+  private fun buildBody(): String {
+    return """
+      $initialBody
+      
+      ${if (allowCrash.isChecked && exception != null) exceptionDetails() else ""}
+      ${if (allowZims.isChecked) zimFiles() else ""}
+      ${if (allowLanguage.isChecked) languageLocale() else ""}
+      ${if (allowDeviceDetails.isChecked) deviceDetails() else ""}
+      ${if (allowFileSystemDetails.isChecked) systemDetails() else ""}
+       """.trimIndent()
+  }
+
+  private fun exceptionDetails(): String {
+    return """
+      Exception Details:
+      ${toStackTraceString(exception!!)}
+      """.trimIndent()
+  }
+
+  private fun zimFiles(): String {
+    val allZimFiles = bookDao.getBooks().joinToString {
+      """
+      ${it.book.getTitle()}:
+      Articles: [${it.book.getArticleCount()}]
+    Parcelable.Creator: [${it.book.getCreator()}]
+      """.trimIndent()
+    }
+    val currentZimFile = zimReaderContainer.zimCanonicalPath
+    return """
+        Current Zim File:
+        $currentZimFile
+        All Zim Files in DB:
+        $allZimFiles
+        
+        """.trimIndent()
+  }
+
+  private fun languageLocale(): String {
+    return """
```

Comment:
you can convert these functions that just return raw strings to expression bodies to make this file a tiny bit slimmer, press `alt+return` with your cursor on `return` and it should give you the option

Reply:
done

Reply:
done